The `no-redeclare` rule seems to not check the module scope.

test.js

```js
export class A {}
export class A {}
```

.eslintrc

```json
{
    "ecmaFeatures": {"classes": true, "modules": true},
    "rules": {"no-redeclare": 2}
}
```

----

```
> eslint -v
v0.24.0
> eslint test.js --reset
```

## Actual

No errors.

## Expected

```
test.js
  2:13  error  A is already defined  no-redeclare

✖ 1 problem (1 error, 0 warnings)
```

----

Maybe, to fix, executes [this block](https://github.com/eslint/eslint/blob/master/lib/rules/no-redeclare.js#L47) when also the modules flag is turned on.

--- a/lib/rules/no-redeclare.js
+++ b/lib/rules/no-redeclare.js
@@ -44,7 +44,7 @@ module.exports = function(context) {
         findVariablesInScope(scope);
 
         // globalReturn means one extra scope to check
-        if (node.type === "Program" && context.ecmaFeatures.globalReturn) {
+        if (node.type === "Program" && (context.ecmaFeatures.globalReturn || context.ecmaFeatures.modules)) {
             findVariablesInScope(scope.childScopes[0]);
         }
     }

--- a/tests/lib/rules/no-redeclare.js
+++ b/tests/lib/rules/no-redeclare.js
@@ -35,6 +35,10 @@ eslintTester.addRuleTest("lib/rules/no-redeclare", {
         { code: "var a = {}; var a = [];", errors: [{ message: "a is already defined", type: "Identifier"}] },
         { code: "var a = function() { }; var a = function() { }", errors: [{ message: "a is already defined", type: "Identifier"}] },
         { code: "var a = function() { }; var a = new Date();", errors: [{ message: "a is already defined", type: "Identifier"}] },
-        { code: "var a = 3; var a = 10; var a = 15;", errors: [{ message: "a is already defined", type: "Identifier"}, { message: "a is already defined", type: "Identifier"}] }
+        { code: "var a = 3; var a = 10; var a = 15;", errors: [{ message: "a is already defined", type: "Identifier"}, { message: "a is already defined", type: "Identifier"}] },
+        { code: "var a; var a;", ecmaFeatures: { modules: true }, errors: [{ message: "a is already defined", type: "Identifier"}] },
+        { code: "export var a; export var a;", ecmaFeatures: { modules: true }, errors: [{ message: "a is already defined", type: "Identifier"}] },
+        { code: "export class A {} export class A {}", ecmaFeatures: { classes: true, modules: true }, errors: [{ message: "A is already defined", type: "Identifier"}] },
+        { code: "export var a; var a;", ecmaFeatures: { modules: true, globalReturn: true }, errors: [{ message: "a is already defined", type: "Identifier"}] }
     ]
 });

